File: src/AnalyticsService/Service/Queries/GetStatsForPeriodQuery.fs
namespace AnalyticsService.Service.Queries

open AnalyticsService.Service.Model
open MediatR
open System

/// A query for obtaining generic stats for a specific time period.
[<Sealed>]
type GetStatsForPeriodQuery(workflowId: Guid, startDate: DateTimeOffset, period: TimeSpan) =
    interface IRequest<GeneralBatchStats>
    member _.WorkflowId = workflowId
    member _.StartDate = startDate
    member _.Period = period

File: src/AnalyticsService/Service/Queries/GetStatsForPeriodQuery.cs
using AnalyticsService.Service.Model;
using MediatR;

namespace AnalyticsService.Service.Queries;

/// <summary>
/// A query for obtaining generic stats for a specific time period.
/// </summary>
internal sealed record GetStatsForPeriodQuery(
    Guid WorkflowId,
    DateTimeOffset StartDate,
    TimeSpan Period
) : IRequest<GeneralBatchStats>;